src/json/serialization.cr

```diff
@@ -117,7 +117,7 @@ module JSON
 
       def self.new(pull : ::JSON::PullParser)
         instance = allocate
-        instance.initialize(pull, nil)
+        instance.initialize(pull_for_json_serializable: pull)
```

Comment:
Might I recommend the usage of double underscore to name the parameters? Macro variables get converted into `__temp_1`, perhaps we should use something similar to reference the private-yet-public interface?

Reply:
IMO that's too much. There's nothing like that used in stdlib ATM.

Reply:
Another option then will be to make initialize private, which can then only be called by this particular `.new` constructor 🤷‍♂️ , thus discouraging externals call to it.